pkg/generator/fluentd/inputs_to_pipeline.go

```diff
@@ -0,0 +1,233 @@
+	Labels     []string
+	Namespaces []string
+	Pipeline   string
+}
+
+type ApplicationsToPipelines []ApplicationToPipeline
+
+func (a ApplicationsToPipelines) Name() string {
+	return "applicationToPipeline"
+}
+
+func (a ApplicationsToPipelines) Template() string {
+	return `{{define "` + a.Name() + `"  -}}
+# Routing Application to pipelines
+<label @_APPLICATION>
+  <filter **>
+    @type record_modifier
+    <record>
+      log_type application
+    </record>
+  </filter>
+  
+  <match **>
+    @type label_router
+	{{- range $index, $a := .}}
+    <route>
+      @label {{$a.Pipeline}}
+      <match>
+        {{- if $a.Namespaces}}
+        namespaces {{comma_separated $a.Namespaces}}
+		{{- end}}
+        {{- if $a.Labels}}
+        labels {{comma_separated $a.Labels }}
+		{{- end}}
+      </match>
+    </route>
+	{{- end}}
+  </match>
+</label>
+{{- end}}`
+}
+
+func SourceTypeToPipeline(sourceType string, spec *logging.ClusterLogForwarderSpec, op Options) Element {
+	srcTypePipeline := []string{}
+	for _, pipeline := range spec.Pipelines {
+		for _, inRef := range pipeline.InputRefs {
+			if inRef == sourceType {
+				srcTypePipeline = append(srcTypePipeline, pipeline.Name)
+			}
+		}
+	}
+	if IsIncludeLegacyForwardConfig(op) && sourceType != logging.InputNameAudit {
+		srcTypePipeline = append(srcTypePipeline, LegacySecureforward)
+	}
+	if IsIncludeLegacySyslogConfig(op) {
+		srcTypePipeline = append(srcTypePipeline, LegacySyslog)
+	}
+	switch len(srcTypePipeline) {
+	case 0:
+		return Nil
+	case 1:
+		return FromLabel{
+			Desc:    fmt.Sprintf("Sending %s source type to pipeline", sourceType),
+			InLabel: helpers.SourceTypeLabelName(sourceType),
+			SubElements: []Element{
+				AddLogSourceType(sourceType),
+				Match{
+					MatchTags: "**",
+					MatchElement: Relabel{
+						OutLabel: helpers.LabelName(srcTypePipeline[0]),
+					},
+				},
+			},
+		}
+	default:
+		return FromLabel{
+			Desc:    fmt.Sprintf("Copying %s source type to pipeline", sourceType),
+			InLabel: helpers.SourceTypeLabelName(sourceType),
+			SubElements: []Element{
+				AddLogSourceType(sourceType),
+				Match{
+					MatchTags: "**",
+					MatchElement: Copy{
+						Stores: CopyToLabels(helpers.LabelNames(srcTypePipeline)),
+					},
+				},
+			},
+		}
+	}
+}
+
+func InputsToPipeline(spec *logging.ClusterLogForwarderSpec, op Options) []Element {
+	return MergeElements(
+		AppToPipeline(spec, op),
+		InfraToPipeline(spec, op),
+		AuditToPipeline(spec, op),
+	)
+}
+
+func AppToPipeline(spec *logging.ClusterLogForwarderSpec, op Options) []Element {
+	userDefined := spec.InputMap()
+	// routed by namespace, or labels
+	routedPipelines := ApplicationsToPipelines{}
+	unRoutedPipelines := []string{}
+	for _, pipeline := range spec.Pipelines {
+		for _, inRef := range pipeline.InputRefs {
+			if input, ok := userDefined[inRef]; ok {
+				// user defined input
+				if input.Application != nil {
+					app := input.Application
+					var a *ApplicationToPipeline = nil
```

Comment:
I would consider initializing the type straight away since that seems to me to be the default expectation and reset it nil if needed